tests/AppBundle/EventListener/CommentNotificationListenerTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\AppBundle\EventListener;
+
+use AppBundle\Entity\Comment;
+use AppBundle\Entity\Post;
+use AppBundle\Entity\User;
+use AppBundle\EventListener\CommentNotificationListener;
+use Prophecy\Argument;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Unit test for the event listener.
+ *
+ * See http://symfony.com/doc/current/book/testing.html#unit-tests
+ *
+ * Execute the application tests using this command (requires PHPUnit to be installed):
+ *
+ *     $ cd your-symfony-project/
+ *     $ ./vendor/bin/phpunit
+ *
+ * @author Michael COULLERET <michael.coulleret@gmail.com>
+ */
+class CommentNotificationListenerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testListenerOnCommentCreated()
```

Comment:
- The registration is done outside the class, it is not possible to test the dispatcher.
- The generator URL is mocked, it is checked that the generation of the link to a comment is executed (here there is no interest to test the dependency, another test already does)
- We test also that the mail is sent, nothing determines that it is an administrator, besides it is precisely: the utility to make functional tests, because here the class it knows only the notion of user (no matter who hides behind)
- Same principle for translation, one already tests its execution

It would also be easy to cut the tests if conditions existed, which is not the case here, the listener is very simple.